src/sql/workbench/parts/query/browser/queryResultsView.ts

```diff
@@ -308,6 +307,7 @@ export class QueryResultsView extends Disposable {
 			dynamicTab.captureState(this.input.state.dynamicModelViewTabsState);
 		});
 
+		[this.resultsTab, this.messagesTab, this.qpTab, this.topOperationsTab, this.chartTab].forEach(t => t.clear());
```

Comment:
it looks like we are doing this on line 298 above.  I think doing this here causes the tabs to lose state when tab switching.